### ISSUE TITLE
Broken detection of indexed source map format

### ISSUE THREAD

**Issue by puzrin** [postcss/postcss#1402]
SM V3 has alernate "indexed" format, useful when you need to concatenate multiple files (with zero cost).

https://github.com/postcss/postcss/blob/daf63bd2504de49fc64198790160a72c2d6eb9a2/lib/previous-map.es6#L149-L152 

`isMap()` checks only default format (while used `sourcemap` 0.6.x supports both).

- Note. If you ever decide to upgrade `sourcemap` version, skip 0.7 and go to 0.8. Because 0.7.3 still has defect - loss of `sourceContent` data.

**Comment by rlidwka** [comment]
Reproduction:

```js
let prev = {
  version:3,
  sections:[
    {
      offset:{line:0,column:0},
      map:{version:3,mappings:'AAAA;AACA;AACA;',sources:['-'],sourcesContent:['body {\nwidth:100%;\n}']}
    }
  ]
}

require('postcss')([]).process('body {\nwidth:100%;\n}',{
  map: {
    // passing source map as an object fails this check:
    // https://github.com/postcss/postcss/blob/daf63bd2504de49fc64198790160a72c2d6eb9a2/lib/previous-map.es6#L151
    prev: prev
    // passing source map as a string works:
    //prev: JSON.stringify(prev)
  }
}).then(r=>console.log(r.map.toJSON()),console.log)
```

**Comment by ai** [comment]
Does Mozilla give a sync API for the new `source-map`? We can’t update because they started to use async API, which does not fit the PostCSS API.

**Comment by puzrin** [comment]
Let's split cases:

1. `sourcemap` v0.6.x has sync api AND supports indexed format.
2. v0.7.x is async and buggy with indexed format (let's skip).
3. v0.8.0-beta0 seems ok and has async api.

So:

- Minimal option is to stay with v0.6.x and fix `isMap()` - that should cost nothing.
- v0.8.x would be fine (much faster)

**Comment by ai** [comment]
We need **sync** API, not async. Can we use 0.8?

If `source-map` supports this format, what do we need to change in PostCSS?

**Comment by puzrin** [comment]
> We need sync API, not async. Can we use 0.8?

Then you should stay with 0.6.1

> If `source-map` supports this format, what do we need to change in PostCSS?

https://sourcemaps.info/spec.html#h.535es3xeprgt

```js
isMap (map) { 
  if (typeof map !== 'object') return false 
  return typeof map.mappings === 'string' ||
         typeof map._mappings === 'string' ||
         Array.isArray(map.sections) // <= missed check
}
```

**Comment by ai** [comment]
Can you send PR with tests to `ose` branch? We will release it in next 8.0 (in a week or two).

**Comment by puzrin** [comment]
Sorry, busy with other projects. I only report to all deps bugs, been found in process of code rewite.

**Comment by ai** [comment]
@rlidwka do you want to help?

**Comment by rlidwka** [comment]
@ai, I've pointed a specific line where the issue is and a code sample for tests. That's the best I can do with the short amount of time at the moment (maybe later on the weekend).

But I think that's one-line change, so someone already familiar with how code and tests work here, will fix this much faster.

Tagging another somewhat relevant issue: https://github.com/postcss/postcss/issues/583

**Comment by rlidwka** [comment]
So I guess we should test for `_sections` as well as `sections` to accomodate mozilla SourceMapConsumer instances from another version of the library - same reason as for `_mappings`?

**Comment by ai** [comment]
Fixed ff9979d

Will be released in 8.0.
